crates/bevy_wgpu/src/lib.rs

```diff
@@ -32,11 +32,45 @@ impl Plugin for WgpuPlugin {
 }
 
 pub fn wgpu_render_system(resources: &mut Resources) -> impl FnMut(&mut World, &mut Resources) {
-    let mut wgpu_renderer = pollster::block_on(WgpuRenderer::new());
+    let options = resources
+        .get_cloned::<WgpuOptions>()
+        .unwrap_or_else(WgpuOptions::new);
+    let mut wgpu_renderer = pollster::block_on(WgpuRenderer::new(options));
     let resource_context = WgpuRenderResourceContext::new(wgpu_renderer.device.clone());
     resources.insert::<Box<dyn RenderResourceContext>>(Box::new(resource_context.clone()));
     resources.insert(SharedBuffers::new(Box::new(resource_context)));
     move |world, resources| {
         wgpu_renderer.update(world, resources);
     }
 }
+
+#[derive(Clone)]
+pub struct WgpuOptions {
+    power_pref: WgpuPowerOptions,
+}
+
+#[derive(Clone)]
+pub enum WgpuPowerOptions {
+    HighPerformance,
+    Adaptive,
+    LowPower,
+}
+
+impl WgpuOptions {
```

Comment:
I think we should just cut this impl in favor of a direct Default impl. In general in Bevy I don't want to use the builder pattern over the Default pattern unless we have a specific reason to make an exception (ex: we need statefulness or special get/set logic).